### ISSUE TITLE
Another IndexError when opening some KAR files

### ISSUE THREAD

**Issue by hyst329** [olemb/mido#93]
Exactly a year after #42 (coincidentally), I've noticed that some of my MIDI karaoke files give an `IndexError` when I'm trying to open them with mido. Further "investigation" showed that these files contains some messages at the very beginning like this:
```
Message:
  00002c: 00  .
-> delta=0
  00002d: ff  .
  00002e: 21  !
  00002f: 01  .
  000030: 00  .
```
which decode to `midi_port`, whose `data` is empty, so 
```python
message.port = data[0] # midifiles/meta.py, line 250 (as of 1.2.5)
```
gives an `IndexError` because `data` is an empty list.

I don't know the purpose of such messages, maybe they're simply markers for some ancient player (these files are mostly Russian and created about 15-20 years ago).
So, I suggest to add a guard to this line:
```python
message.port = data[0] if len(data) else 0 # now loading works
```
(Sorry for being too lazy for a pull request, and sorry another time for the bad English (my native language is Russian).)

**Comment by olemb** [comment]
No problem at all. A pull request is not needed for this small change. I
need do some tests with other files to make sure it doesn't break anything
but if this works out I will include it in the next version. I'll get back
to you when I've run the tests.

**Comment by olemb** [comment]
I have made the changes and pushed them to my master branch. Could you test and see if it works for your files?

**Comment by hyst329** [comment]
@olemb OK, I'll test it as soon as possible (probably around 21:00 UTC+3 today), but if you've changed only this line, I could definitely say that it'll work (because I've tested it by modifying the file in 1.2.5 release)

**Comment by olemb** [comment]
Sounds good!

I also added some tests to ensure that it keeps working in the future.

**Comment by olemb** [comment]
Since I didn't hear back from you I'm assuming it works. I just released 1.2.6 which includes your fix. As you said it should work but if you have any trouble just let me know and I will reopen the issue.

**Comment by hyst329** [comment]
@olemb Sorry, I've lost my connection to the Internet. Yes, I can assure you that it works.

**Comment by olemb** [comment]
Ah, I see. No problem. Thanks for reporting this.

**Comment by hyst329** [comment]
@olemb Thank you too (for fixing).
